src/data-model-providers/codegen/CodegenDataModelProvider.cpp

```diff
@@ -164,6 +163,9 @@ CHIP_ERROR CodegenDataModelProvider::Startup(DataModel::InteractionModelContext
     });
 }
 
+// Command validation queries this provider for metadata before InvokeCommand runs, so the dispatch sequence below relies on the
+// same catalog of clusters and handlers that validation already approved.  That keeps the privilege and flag checks performed by
+// `ValidateCommandCanBeDispatched` in sync with the handler that ultimately processes the request.
```

Comment:
I don't understand what this comment is trying to say and why it's specific to CodegenDataModelProvider.

As in: why is this comment here?